### PR TITLE
Login - WIP

### PR DESCRIPTION
**Pull request by cammac60**
### This PR: 
- Adds and styles the login component.
- Adds a post request for user login and retrieves the server's response. 
- Adds an action creator and reducer for login. 
- Adds a back button to the login page. 

### Where to look: 
- `src/actions/index.js` for the action.
- `src/apiCalls/apiCalls.js` for the API call.
- Line 17 of `src/components/App/App.js` for the new redirection.
- New component: `src/containers/Login/Login.js`
- Styles: `src/containers/Login/Login.scss`
- `src/reducers/user.js` for the new reducer.

### Notes:
- The store is not being updated with the reducer. I am unsure why this is happening as the console log on line 72 of `src/containers/Login/Login.js` is returning the correct object. I have tried changing everything I can think of but state is still the same. I'm sure it's something simple that I have just failed to see but I think I have tunnel vision from looking at this so long. 

- We still need to add functionality that happens after the login is successful (ie: reload movies page w/ user's reviews, add logout functionality, and clear the local state of the form component).

### CODE REVIEW

**Comment by RayRedGoose** [comment]
I found why user wasn't pushed to the store: you don't put action creator to dispatch, and use exported function instead of prop function.

File: src/containers/Login/Login.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    event.preventDefault();
+    if (this.validateForm()) {
+      let user = {
+        email: this.state.email,
+        password: this.state.password
+      };
+      try {
+        const result = await postSignIn(user)
+        console.log(addUser(result.user).user)
+        addUser(result.user)
```
Comment (RayRedGoose) [inline src/containers/Login/Login.js]: should be `this.props.addUser` instead of just `addUser`

File: src/containers/Login/Login.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    }
+  }
+
+}
+
+export const mapDispatchToProps = dispatch => (
+  {
+    addUser
+  }
+)
```
Comment (RayRedGoose) [inline src/containers/Login/Login.js]: you don't add this action creators to dispatch, and it can't to put it as correct prop function

Comment (cammac60) [reply src/containers/Login/Login.js]: @RayRedGoose Thank you! Can you please elaborate on what you mean by this, I'm unsure if you're saying I need to change anything here. Your other comment makes total sense though.

**Comment by cammac60** [comment]
@RayRedGoose @andrewckinstler I just made a push that adds the logout functionality and fixes merge conflicts. However, there is still a bug with the way the login button is setup. 

Essentially, because the link element is within the button (line 71 of `Login.js`) you have to click the text directly for the redirection to happen. I have tried styling the link the same as the button and well as trying to expand the link to fill the entire button but neither have worked. I can look into this more tomorrow but please let me know if you have any suggestions.

**Comment by cammac60** [comment]
@RayRedGoose @andrewckinstler  I think we are finally good to merge this but you may want to pull down to make sure that everything is working for you too. 

As per Raisa's suggestion I still need to add profile info to the header on login but I can make a new branch for that since it has more to do with the header component.

File: src/containers/Login/Login.js [hunk 3]
```diff
@@ -62,7 +62,7 @@ export class Login extends Component {
         email: '',
         password: ''
       });
-      this.props.history.push('/');
+      this.props.history.push('/ratings');
```
Comment (RayRedGoose) [inline src/containers/Login/Login.js]: first one was correct. Redirect to ratings should be after user see all movies. There is should be button for redirect on all movies page

**Pull request merged**